pulpcore/pulpcore/app/viewsets/content.py

```diff
@@ -61,6 +61,22 @@ def create(self, request):
         headers = self.get_success_headers(serializer.data)
         return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
 
+    def destroy(self, request, pk):
+        """
+        Remove Content unit only if it is not associated with a repository version
+        """
+        repository_content_set = RepositoryContent.objects.filter(content__pk=pk,
+                                                                  version_removed=None)
+        if repository_content_set:
+            msg = _('The Content unit cannot be deleted because it is associated with a repository '
+                    'version, which is immutable.')
+            data = {'detail': msg}
+            return Response(data, status=status.HTTP_409_CONFLICT)
+        else:
```

Comment:
`else` might not really be needed here

Reply:
we probably should, it just needn't be under the `else` statement as there's the `return` on previous line but matter of taste I guess...